lib/beaker/hypervisor/docker.rb

```diff
@@ -65,6 +65,19 @@ def provision
         host['docker_container'] = container
         host['docker_image'] = image
       end
+
+      hack_docker_etc_hosts @hosts, @options
+
+    end
+
+    def hack_docker_etc_hosts hosts, opts
+      etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
+      hosts.each do |host|
+        etc_hosts += "#{host['docker_container'].json["NetworkSettings"]["IPAddress"]}\t#{host.name}\n"
```

Comment:
I'll gather some data, attempt using `hack_etc_hosts` again, and post back here :)